### DISCUSSION TITLE
Beacons Monitoring OnStatusChanged not firing

### DISCUSSION THREAD

**Discussion by RapAst51** [shinyorg/shiny#884] [Q&A]
Hi 
I follow this guide https://allanritchie.com/posts/shiny20

I install the beacon stuff but when i tried to enter or exit beacon radius the OnStatusChanged function doesnt fired.

This is my code:
ShinyStartup:

`   
public class ShinyStart : ShinyStartup
    {
        public override void ConfigureServices(IServiceCollection services, IPlatform platform)
        {

            services.UseNotifications();
            services.UseBeaconMonitoring<MyBeaconMonitorDelegate>(new Shiny.Beacons.BeaconMonitorConfig().UseEstimote());

        }
    }

`

MyBeaconMonitorDelegate

`
   public class MyBeaconMonitorDelegate : IBeaconMonitorDelegate
    {
     
        public async Task OnStatusChanged(BeaconRegionState newStatus, BeaconRegion region)
        {

    
        }

    }
`

App -> Beacon start monitoring

`  
  IBeaconMonitoringManager? monitorManager =  ShinyHost.Resolve<Shiny.Beacons.IBeaconMonitoringManager>();

            monitorManager.RequestAccess();

            const string EstimoteUuid = "12345678-9123-4567-8912-345678912345";
            var guid = Guid.Parse(EstimoteUuid);

            monitorManager.StartMonitoring(new BeaconRegion("Bacon 1", guid));

`

**Comment by aritchie** [comment]
File an issue with a FULL reproducible sample

**Reply by RapAst51** [reply]
Here is a reproductible sample 

https://www.mediafire.com/file/yhxm2odcbej023h/Sample.rar/file

**Reply by aritchie** [reply]
Please ensure you file in the proper issue section next time.  https://github.com/shinyorg/shiny/issues/new/choose

In your startup, you've got the estimote scan filter being installed.  Do you have estimote beacons?

**Reply by aritchie** [reply]
You should start with ranging to ensure your beacons are being scanned properly than move to monitoring.

**Reply by RapAst51** [reply]
Sorry about filing in wrong section.

I dont use EstimoteBeacons, thats right. Its my first time working with beacons btw.

 `         services.UseBeaconMonitoring<MyBeaconMonitorDelegate>(new Shiny.Beacons.BeaconMonitorConfig().UseEstimote());`

But i can't put only:
         ` services.UseBeaconMonitoring<MyBeaconMonitorDelegate>(); `
Because it needs the BeaconMonitorConfig object.

Can you help me with that?

**Reply by aritchie** [reply]
Not really - you need to figure out what your beacon uses for a service UUID so that it can be seen when the app is fully in the background

**Reply by RapAst51** [reply]
Hi,
I followed your recomendation and i first used the ranging method. It works, but when i try to use the monitoring method but it still dont fire the  OnStatusChanged() method in the IBeaconMonitorDelegate.
I configured the config file with the manufacturer UUID.
It needs something special for it?
